core/src/main/java/com/yahoo/imapnio/async/data/FetchResult.java

```diff
@@ -0,0 +1,33 @@
+package com.yahoo.imapnio.async.data;
+
+import java.util.List;
+
+import javax.annotation.Nonnull;
+
+import com.sun.mail.imap.protocol.IMAPResponse;
+
+/**
+ * This class provides the list of IMAP response from fetch command response.
+ */
+public class FetchResult {
```

Comment:
would be interested to know why there is a new FetchResult class. why we cant use ImapAsyncResponse?

Reply:
Because I want to have a parser which can cast IMAPResponse to FetchResponse through ImapResponseMapper. Perhaps, I should handle the parsing in the FetchResult constructor instead of ImapResponseMapper. Therefore, users can cast directly in FetchResult instead of ImapResponseMapper to avoid the same implementation as ImapAsyncResponse.